aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -920,20 +914,15 @@ Tensor fmod(const Tensor& self, const Tensor & other) {
 }
 
 Tensor fmod(const Tensor& self, Scalar other) {
-  Tensor other_tensor = wrapped_scalar_tensor(other);
-  // FIXME: 'other' is converted to match the dtype of 'self' to retain
-  //   BC with TH, but in the future, we should use normal type promotion,
-  //   like in numpy
-  // Issue #47779: https://github.com/pytorch/pytorch/issues/47779
-  return native::fmod(self, other_tensor.to(self.dtype()));
+  return native::fmod(self, wrapped_scalar_tensor(other));
 }
 
 Tensor& fmod_(Tensor& self, const Tensor& other) {
   return native::fmod_out(self, self, other);
 }
 
 Tensor& fmod_(Tensor& self, Scalar other) {
-  return native::fmod_out(self, self, other);
+  return native::fmod_(self, wrapped_scalar_tensor(other));
```

Comment:
Interesting. So fmod doesn't support scalar x tensor inputs?

Reply:
It's same as `div` and `mul`, only supporting tensor x scalar inputs.
If we want to support, we may need to let most of operators supporting this behavior.

Reply:
Thank you for verifying. Supporting scalar x tensor with more operators is interesting. I agree with you that we should consider a change like that holistically.